### PR TITLE
Fix TypeError - no implicit conversion of nil into String.

### PR DESCRIPTION
**Pull request by yanzhiwei147**
Fix [#6381](https://github.com/CocoaPods/CocoaPods/issues/6381)

### CODE REVIEW

**Comment by yanzhiwei147** [comment]
The unit test spec with fail is incorrect. we have one more repo can contain the Reachability, that use `pod search Reachability` we show below expect:

```
-> Reachability (3.2)
   ARC and GCD Compatible Reachability Class for iOS and OS X. Drop in replacement for Apple Reachability.
   pod 'Reachability', '~> 3.2'
   - Homepage: https://github.com/tonymillion/Reachability
   - Source:   https://github.com/tonymillion/Reachability.git
   - Versions: 3.2, 3.1.1, 3.1.0, 3.0.0 [master repo] - 3.2, 3.1.1, 3.1.0, 3.0.0 [sdp-mirrors-specs repo]
```

but now i get below(It miss the private repo results):

```
-> Reachability (3.2)
   ARC and GCD Compatible Reachability Class for iOS and OS X. Drop in replacement for Apple Reachability.
   pod 'Reachability', '~> 3.2'
   - Homepage: https://github.com/tonymillion/Reachability
   - Source:   https://github.com/tonymillion/Reachability.git
   - Versions: 3.2, 3.1.1, 3.1.0, 3.0.0 [master repo]
```

**Comment by benasher44** [comment]
Is that a local failure you're referring to? The spec failing on travis has different output.

**Comment by benasher44** [comment]
Thanks for the PR!

**Comment by segiddins** [comment]
Can you please add a test for the issue this is fixing? Thanks!

**Comment by yanzhiwei147** [comment]
@segiddins It's ok.

**Comment by yanzhiwei147** [comment]
@benasher44 Yeah,you can look the new PR

**Comment by dnkoutso** [comment]
@yanzhiwei147 is this PR attempting to fix two things or just one thing? The PR description seems like its addressing a single issue but you are also including a performance issue?

**Comment by yanzhiwei147** [comment]
This PR do the two things as you can see:

1.  Resolve dependency performance. but merged in [This PR](https://github.com/CocoaPods/Core/commit/559584456461e929871e16596971d1b384c86e7f).
2. Resolve search issues.

**Comment by yanzhiwei147** [comment]
This PR can be merged?

**Comment by dnkoutso** [comment]
@yanzhiwei147 please rebase first

**Comment by yanzhiwei147** [comment]
@dnkoutso It's ok now.

**Comment by segiddins** [comment]
I'm still seeing merge commits, so it doesn't look as if it's rebased

File: CHANGELOG.md [hunk 1]
```diff
@@ -189,6 +169,10 @@
   helpful error.  
   [Samuel Giddins](https://github.com/segiddins)
   [#328](https://github.com/CocoaPods/Core/issues/328)
+    
+* Improve the performance of resolve dependency.
```
Comment (segiddins) [inline CHANGELOG.md]: Wrong section

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,29 +8,9 @@
 
 ##### Bug Fixes
 
-* None.  
-
-
-## 1.2.1.rc.1 (2017-04-05)
-
```
Comment (segiddins) [inline CHANGELOG.md]: All of this shouldn't be deleted

**Comment by dnkoutso** [comment]
@yanzhiwei147 can you clean up this PR please?

**Comment by yanzhiwei147** [comment]
@dnkoutso Yes.

**Comment by yanzhiwei147** [comment]
It can be merged now?

**Comment by dnkoutso** [comment]
@yanzhiwei147 please rebase/squash and resolve conflict

**Comment by dnkoutso** [comment]
do we need this PR?

**Comment by dnkoutso** [comment]
Going to close this for now. @yanzhiwei147 re-open it if you can finish it.

**Comment by yanzhiwei147** [comment]
@dnkoutso Sorry,it can resolved today.

**Pull request closed without merging**